Friendli Integration

kooyunmo opened the pull request:
# Description

This PR adds [Friendli](https://friendli.ai/) integration for `Friendli` LLM.
Friendli offers a comprehensive generative AI platform offering endpoint services for easy access to AI models like Llama-2 and Gemma. This integration supports chat and completions APIs for text generation models available in Friendli. The integration is implemented with [Friendli SDK](https://github.com/friendliai/friendli-client) which can be installed with `pip install friendli-client`.

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

logan-markewich commented on llama-index-integrations/llms/llama-index-llms-friendli/pyproject.toml:
```diff
@@ -0,0 +1,47 @@
+check-filenames = true
+check-hidden = true
+skip = "*.csv,*.html,*.json,*.jsonl,*.pdf,*.txt,*.ipynb"
+
```
Can you add this here?

```
[tool.llamahub]
contains_example = false
import_path = "llama_index.llms.gemini"

[tool.llamahub.class_authors]
Friendli = "kooyunmo"
```

logan-markewich commented on docs/examples/llm/friendli.ipynb:
Do you mind adding a link to this in `docs/module_guides/llms/modules.md` ?

kooyunmo commented:
@logan-markewich Thanks for the quick review! I've addressed your comments. Please let me know if there are any more things to do. I'm keeping a close eye on this PR.

kooyunmo commented:
@logan-markewich Now it passes all checks. Could you please confirm the changes?
